### DISCUSSION TITLE
Trucate collection functionality

### DISCUSSION THREAD

**Discussion by Meldiron** [appwrite/appwrite#2386] [Ideas]
_Idea by Tanner Meade on Discord_

Let's add a button to database collection's settings just above "Delete collection" saying "Truncate collection". This function could use all of the existing APIs and in theory does not require any new endpoints. The function would iterate over all documents in the collection and delete them.

**Comment by Meldiron** [comment]
Unrelated conversation, but someone else pointed out that there is no button to delete all users, which is similar to truncating a database. With that said, this feature could also include:

- Delete all users
- Delete all files
- Delete all function tags
And so on

**Comment by eldadfux** [comment]
Would you see this as an additional flag in the existing endpoints, how should we name it?
Another question is what should be the default behavior, a complete deletion or just truncating the data?

**Reply by amosogra** [reply]
If it is possible to add both behaviors I think it would be nice... Also there should be a form of authorization/permission so as not to delete project sensitive data mistakenly

**Reply by NeonSpork** [reply]
Just an idea regarding the name, it could be named something like `deleteDocuments('[COLLECTION_ID]')` or `removeDocuments('[COLLECTION_ID]')` to mirror the name of for example `listDocuments('[COLLECTION_ID]')` already in the database functions? Maybe the "remove" variant since there's already a `deleteDocument()` that removes a single entry.

I guess the option exists to iterate over the entire collection and delete each entry with the latter example, but such a function would be less verbose code in the application, I'm sure.

**Reply by eldadfux** [reply]
@NeonSpork most of the Appwrite naming convention and whether we use a new method or an existing one are derived from the HTTP API implementation. For the functionality, the REST API might look like:
`DELETE /v1/database/collections/[id]` - for both options with a param called `truncate` which can be true or false.

Another options is to have something like:
`DELETE /v1/database/collections/[id]/truncate`

I think the first one makes more sense as its more consistent with the rest of the endpoints in REST API.

**Comment by kodumbeats** [comment]
I personally see this as a CLI command either:

- to be executed as a task `docker compose exec appwrite truncate $ID`
- a tool in the Appwrite CLI `appwrite database truncateCollection --collectionID=$ID`

Is the Appwrite console wired up to execute commands in `app/tasks`?

**Reply by hey-nicolasklein** [reply]
I would love a feature like this 👍
